airflow/models/dag.py

```diff
@@ -1976,7 +1981,7 @@ def create_dagrun(
         run_id: Optional[str] = None,
         start_date: Optional[datetime] = None,
         external_trigger: Optional[bool] = False,
-        conf: Optional[dict] = None,
+        conf: Optional[dict] = {},
```

Comment:
This will suffer from one of the common Python gotcha's about Mutable default args

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
@kaxil I think the way we are using the `conf` variable, it would be safe from this gotcha, for example see this
```
def safe_func(args={}):
    print(args)

def prob_func(arg1={}, arg2={}):
    arg2.update(arg1)
    print(arg2)

if __name__ == '__main__':
    safe_func(args={'a': 1})
    safe_func()
    safe_func(args={'a': 2})

    prob_func(arg1={'a': 1})
    prob_func()
    prob_func(arg1={'a': 2})
```
generate outputs:
```
{'a': 1}
{}
{'a': 2}
{'a': 1}
{'a': 1}
{'a': 2}
```

Reply:
This is only a problem if/when `dagrun.conf` is ever edited in place (by adding keys etc) -- and given we can't tell where `dagrun.conf` might be used elsewhere in the codebase it would be safer to change this, because while it might not cause problems _right now_ it might bite us later.